### PR TITLE
Sdk v1.23.0/golang 1.18 upgrade

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: config/crd/bases/osp-director.openstack.org_openstackbackuprequests.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.8.0
+    controller-gen.kubebuilder.io/version: v0.9.0
```
Comment (abays) [inline config/crd/bases/osp-director.openstack.org_openstackbackuprequests.yaml:6]: Didn't we run into some sort of issue with `v0.9.0` with Podified such that we immediately bumped to `v0.9.2`?  If so, just wondering if we want to do that here too

**Comment by dprince** [comment]
Pushing this to address the Prow golangci failure: https://github.com/openshift/release/pull/31969

File: config/crd/bases/osp-director.openstack.org_openstackbackuprequests.yaml (hunk 1 above)
Comment (dprince) [reply config/crd/bases/osp-director.openstack.org_openstackbackuprequests.yaml:6]: I'll bump it. Wasn't strictly required for this update but yes we've done it elsewhere and won't harm anything.

Comment (dprince) [reply config/crd/bases/osp-director.openstack.org_openstackbackuprequests.yaml]: I had a cached version in bin/controller-gen. Removed and re-ran make. All good now I think

File: Makefile [hunk 2]
```diff
@@ -242,7 +245,7 @@ ifeq (,$(shell which opm 2>/dev/null))
 	set -e ;\
 	mkdir -p $(dir $(OPM)) ;\
 	OS=$(shell go env GOOS) && ARCH=$(shell go env GOARCH) && \
-	curl -sSLo $(OPM) https://github.com/operator-framework/operator-registry/releases/download/v1.19.1/$${OS}-$${ARCH}-opm ;\
```
Comment (olliewalsh) [inline Makefile:245]: Did you mean to drop the ;\

Comment (dprince) [reply Makefile:245]: @olliewalsh I did not. Thanks! Should be fixed now.

**Review by olliewalsh — APPROVED** [review]
/lgtm

**Pull request merged**